python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -1125,7 +1141,7 @@ def __post_init__(self) -> None:
             old, new, _ = self.options
             # TODO: perhaps polars should validate renaming in the IR?
             if len(new) != len(set(new)) or (
-                set(new) & (set(self.df.schema.keys() - set(old)))
+                set(new) & (set(self.df.schema.keys()) - set(old))
```

Comment:
benign typo

Reply:
I'm confused as to how this was even working before. self.df.schema.keys() would be an iterable, not a set, so how was `self.df.schema.keys() - set(old)` valid?

Reply:
It's a dict_keys and evidently `__rsub__` works